vidarr/wdl.py

```diff
@@ -0,0 +1,237 @@
+import argparse
+
+import json
+import os
+import sys
+from typing import Dict, Any
+import WDL
+
+_output_mapping = [
+    (WDL.Type.File(),
+     "file"),
+    (WDL.Type.Array(
+        WDL.Type.File()),
+     "files"),
+    (WDL.Type.Pair(
+        WDL.Type.File(),
+        WDL.Type.Map(
+            (WDL.Type.String(), WDL.Type.String()))),
+     "file-with-labels"),
+    (WDL.Type.Pair(
+        WDL.Type.Array(
+            WDL.Type.File()),
+        WDL.Type.Map(
+            (WDL.Type.String(),
+             WDL.Type.String()))),
+     "files-with-labels"),
+    (WDL.Type.Boolean(),
+     "quality-control"),
+    (WDL.Type.File(
+        optional=True),
+     "optional-file"),
+    (WDL.Type.Array(
+        WDL.Type.File(),
+        optional=True),
+     "optional-files"),
+    (WDL.Type.Array(
+        WDL.Type.File(),
+        nonempty=True),
+     "optional-files"),
```

Comment:
Wouldn't non-empty be non-optional?

Reply:
Oops. Yes! I' had to add some additional cases.

Reply:
So to confirm, some of these can be both optional and nonempty?

Reply:
I think so. If it's non-empty and mandatory, there will definitely be a file, which is what we want for considering something mandatory. Any other case means it might no have something, right?

Reply:
That makes sense!